Classes/PlusOnes.lua

```diff
@@ -2,126 +2,970 @@
+            return true;
+        end
+    end
 
-    return DB.PlusOnes[playerName];
+    return false;
 end
 
---- Deduct a plus one for the given player name and return the current plusOne value
+--- Mark a given player as "trusted"
 ---
 ---@param playerName string
----@return number
-function PlusOnes:deduct(playerName)
-    GL:debug("PlusOnes:deduct");
+---@return void
+function PlusOnes:markPlayerAsTrusted(playerName)
+    playerName = strtrim(playerName);
 
-    playerName = GL:normalizedName(playerName);
+    if (GL:empty(playerName)
+        or self:playerIsTrusted(playerName)
+    ) then
+        return;
+    end
 
-    if (not DB.PlusOnes[playerName]) then
-        DB.PlusOnes[playerName] = 0;
+    local trustedPlayerCSV = GL.Settings:get("PlusOnes.automaticallyAcceptDataFrom", "");
+    if (GL:empty(trustedPlayerCSV)) then
+        GL.Settings:set("PlusOnes.automaticallyAcceptDataFrom", playerName);
+        return
     else
-        -- Make sure the plusOne doesn't end up below 0
-        DB.PlusOnes[playerName] = max(DB.PlusOnes[playerName] - 1, 0);
+        local TrustedPlayers = GL:strSplit(trustedPlayerCSV, ",");
+        tinsert(TrustedPlayers, playerName);
+        GL.Settings:set("PlusOnes.automaticallyAcceptDataFrom", table.concat(TrustedPlayers, ","));
     end
+end
 
-    self:triggerChangeEvent();
+--- Remove a player from the list of "trusted" players
+---
+---@param playerName string
+---@return void
+function PlusOnes:removePlayerFromTrusted(playerName)
+    playerName = strtrim(playerName);
+
+    -- No point removing the player if he's not trusted in the first place
+    if (GL:empty(playerName)
+        or not self:playerIsTrusted(playerName)
+    ) then
+        return;
+    end
+
+    local trustedPlayerCSV = GL.Settings:get("PlusOnes.automaticallyAcceptDataFrom", "");
+    local TrustedPlayers = GL:strSplit(trustedPlayerCSV, ",");
+    local NewTrustedPlayers = {};
+    local normalizedName = GL:normalizedName(playerName);
 
-    return DB.PlusOnes[playerName];
+    for _, trustedPlayer in pairs(TrustedPlayers) do
+        if (GL:normalizedName(trustedPlayer) ~= normalizedName) then
+            tinsert(NewTrustedPlayers, trustedPlayer);
+        end
+    end
+
+    GL.Settings:set("PlusOnes.automaticallyAcceptDataFrom", table.concat(NewTrustedPlayers, ","));
 end
 
---- Set a given player's plusOne value to 0
+--- Check whether plus ones are enabled
 ---
----@param playerName string
----@return number
-function PlusOnes:setToZero(playerName)
-    GL:debug("PlusOnes:setToZero");
+---@return boolean
+function PlusOnes:enabled()
+    GL:debug("PlusOnes:enabled");
+
+    return GL.Settings:get("PlusOnes.enabled", false);
+end
+
+--- Check whether there are plus ones available
+---
+---@return boolean
+function PlusOnes:available()
+    GL:debug("PlusOnes:available");
+
+    --return GL:higherThanZero(DB:get("PlusOnes.MetaData.updatedAt", 0)); --not sure this check is even needed for plus ones
```

Comment:
You want to keep this is place or want to add a `---@todo: ` and leave a comment for me so I can check it out later?

Reply:
I really don't think it's necessary for plus ones since I allow broadcasting empty data when a clear occurs.  I left this function in to not break the code where it is called.